### DISCUSSION TITLE
"\\wsl.localhost not accessible" *fixed*

### DISCUSSION THREAD

**Discussion by james28909** [microsoft/WSL#7742] [General]
Ok, so I just read a few threads about how to fix this error. None of them worked for me, my reg was already good to go and everything else seemed to be in working order except for accessing this server at \\\\wsl.localhost (the Linux shortcut in navigation pane).

Well guess what? 

### **TURN ON NETWORK DISCOVERY!!!! xD**

Did that for private networks, and now it is working seamlessly. I can finally click “Linux” in explorer quick access again and see all my distros.

**Comment by thistlillo** [comment]
It dos not work any more on Windows 11 Pro. Or, at least, it did not work on mine, together with the solutions published here: https://www.tenforums.com/tutorials/49652-turn-off-network-discovery-windows-10-a.html (for Win 10).

**Reply by Frushtration** [reply]
Hey, for all the Windows 11 Pro People,

[This link ](https://github.com/microsoft/WSL/issues/4027#issuecomment-496628274) resolved my issues. 

As a quick note for clarity, change the keys of "ProviderOrder" in both the "Order" and "HwOrder" sections. Even If you see the "P9NP" if that line, you need to move that key to the front, or else it won't work. Mine "P9NP" key wasn't missing, it was just at the back, which caused this issue.

Hope that helps!

**Comment by HectorOses** [comment]
Thanks. I tried, but that didn't make it for me. Any other ideas?
PS C:\Windows\system32> reg query HKLM\SYSTEM\CurrentControlSet\Control\NetworkProvider /s

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\HwOrder
    ProviderOrder    REG_SZ    P9NP,cbfsconnect2017,RDPNP,LanmanWorkstation,webclient

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\Order
    ProviderOrder    REG_SZ    P9NP,cbfsconnect2017,RDPNP,LanmanWorkstation,webclient

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\ProviderOrder
    LanmanWorkstation    REG_DWORD    0x7d0
    P9NP    REG_DWORD    0x1f4
    RDPNP    REG_DWORD    0x3e8
    webclient    REG_DWORD    0xbb8
    cbfsconnect2017    REG_DWORD    0xfa

PS C:\Windows\system32>

**Comment by HectorOses** [comment]
Finally I got it working doing this:

At first I was getting no access, then making this changes in registry made it:
1. have P9NP first for both HwOrder and Order
2. remove from both the "cbfsconnect2017"

```
PS C:\Windows\system32> reg query HKLM\SYSTEM\CurrentControlSet\Control\NetworkProvider /s

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\HwOrder
    ProviderOrder    REG_SZ    P9NP,RDPNP,LanmanWorkstation,webclient

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\Order
    ProviderOrder    REG_SZ    P9NP,RDPNP,LanmanWorkstation,webclient

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\ProviderOrder
    LanmanWorkstation    REG_DWORD    0x7d0
    P9NP    REG_DWORD    0x1f4
    RDPNP    REG_DWORD    0x3e8
    webclient    REG_DWORD    0xbb8
    cbfsconnect2017    REG_DWORD    0xfa

PS C:\Windows\system32>
```

After reboot it gets somehow reset, cbfsconnect2017 becomes first again, but now P9NP is second, but access works

```
PS C:\Windows\system32> reg query HKLM\SYSTEM\CurrentControlSet\Control\NetworkProvider /s

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\HwOrder
    ProviderOrder    REG_SZ    cbfsconnect2017,P9NP,RDPNP,LanmanWorkstation,webclient

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\Order
    ProviderOrder    REG_SZ    cbfsconnect2017,P9NP,RDPNP,LanmanWorkstation,webclient

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\ProviderOrder
    LanmanWorkstation    REG_DWORD    0x7d0
    P9NP    REG_DWORD    0x1f4
    RDPNP    REG_DWORD    0x3e8
    webclient    REG_DWORD    0xbb8
    cbfsconnect2017    REG_DWORD    0xfa

PS C:\Windows\system32>
```

**Reply by paulsybrandy** [reply]
That was the final trick my friend! My savior. Thankies.

**Reply by niladrishekhardutta** [reply]
Thanks. This trick worked.

**Comment by StuCjk** [comment]
same error, same fix, thanks! XD

**Comment by tehido** [comment]
upvote, thanks

**Comment by mucsi96** [comment]
For the following solution resolved the issue: https://github.com/microsoft/WSL/issues/7499#issuecomment-937328189

**Comment by NickDrouin** [comment]
I tried a few of the above, it did not work.  I removed the Box drive app, which seemed linked to the installation of `cbfsconnect2017`.

With Box drive removed, I immediately gained back my distros under the Linux menu of File Explorer.  The "\wsl.localhost is not accessible." is gone.


Ref: Box support thread which discusses other issues with driveio: 
https://support.box.com/hc/en-us/community/posts/19018028785427-BOX-Drive-installs-cbfsconnect2017-sys-which-disables-support-for-Microsoft-s-Bypass-IO-directstorage

**Comment by LaserKaspar** [comment]
I am having the same issue while trying to open it via the sidepanel:
![image](https://github.com/user-attachments/assets/d58449f8-94a5-4877-9afa-d73c6eaa6021)

![image](https://github.com/user-attachments/assets/ad2dc448-a76b-491f-ab15-c344a23b092a)

When I open it via ```explorer.exe .``` from wsl it works without issues.

I tried changing the Provider Order/HwOrder and weights but I still have the same issue. 
![image](https://github.com/user-attachments/assets/ca35d87a-5fc9-4c84-a1f0-2bcbd3f6c5dd)
![image](https://github.com/user-attachments/assets/48b5263f-f493-433a-8b76-bb3aa6e02de7)
Any other ideas on how to fix this?

In case its related to Network settings:
![image](https://github.com/user-attachments/assets/733fee19-29b7-4d4d-82e4-6356eb946cf9)
